libnetwork/netutils/utils_linux.go

```diff
@@ -31,7 +31,7 @@ func CheckRouteOverlaps(toCheck *net.IPNet) error {
 		return err
 	}
 	for _, network := range networks {
-		if network.Dst != nil && NetworkOverlaps(toCheck, network.Dst) {
+		if network.Dst != nil && network.Scope == netlink.SCOPE_LINK && NetworkOverlaps(toCheck, network.Dst) {
```

Comment:
Curious; should this check only be performed in this case, or also in other cases where `NetworkOverlaps` is used? (mostly wondering if this should be done in `NetworkOverlaps()` itself)?

Reply:
I think adding it to `NetworkOverlaps()` risks surprising someone else glancing at the code, the networks are overlapping so `NetworkOverlaps()` is performing exactly as expected, but during this specific comparison we only care about specific routes.
So without renaming it I'd be reluctant to put it there

Though looking at `deleteInterfaceBySubnet()` from `ov_utils.go` maybe it should be added, unfortunately I don't know anything about the overlay network so I have some code and documentation reading before I dare comment on that.
However, `checkOverlap()` from `ov_network.go` which uses `NetworkOverlaps()` should probably inherit this.

Reply:
Having had a better look, I don't think it should be added to `NetworkOverlaps()`, this specific check is only being done in `CheckRouteOverlaps()` the other places where this is used it's being used to literally compare networks.

So for direct usages of this, all looks fine:
- `deleteInterfaceBySubnet()` from `overlay` is checking subnets
- `CheckNameserverOverlaps()` from `netutils` checks if a given IP is in a specific subnet
- `CheckRouteOverlaps()` from `netutils` is the only one looking at routes and that's being addressed in this PR

I've also pushed updated tests that verify the routes (checking both that the overlap works and that it does not overlap with routes with the wrong scope)